src/modules/75-cd/components/EnvironmentGroups/EnvironmentGroupsFilters/EnvironmentGroupsFilters.tsx

```diff
@@ -0,0 +1,251 @@
+      orgIdentifier,
+      projectIdentifier,
+      type: 'EnvironmentGroup'
+    }
+  })
+
+  const { data: environmentsResponse, loading: isFetchingEnvironments } = useGetEnvironmentListForProject({
+    queryParams: { accountId, orgIdentifier, projectIdentifier },
+    lazy: isFiltersDrawerOpen
+  })
+
+  const appliedFilter =
+    queryParams.filterIdentifier && queryParams.filterIdentifier !== UNSAVED_FILTER_IDENTIFIER
+      ? getFilterByIdentifier(queryParams.filterIdentifier, filters)
+      : queryParams.filters && !isEmpty(queryParams.filters)
+      ? {
+          name: UNSAVED_FILTER,
+          identifier: UNSAVED_FILTER_IDENTIFIER,
+          filterProperties: queryParams.filters,
+          filterVisibility: undefined
+        }
+      : null
+
+  const { envGroupName, description, envGroupTags, envIdentifiers } =
+    (appliedFilter?.filterProperties as EnvironmentGroupFilterFormType) || {}
+  const { name = '', filterVisibility, identifier = '' } = appliedFilter || {}
+  const fieldToLabelMapping = React.useMemo(
+    () =>
+      new Map<string, string>([
+        ['envGroupName', getString('common.environmentGroup.label')],
+        ['description', getString('description')],
+        ['envGroupTags', getString('tagsLabel')],
+        ['envIdentifiers', getString('environments')]
+      ]),
+    [getString]
+  )
+
+  const filterWithValidFields = removeNullAndEmpty(
+    pick(flattenObject(appliedFilter?.filterProperties || {}), ...fieldToLabelMapping.keys())
+  )
+
+  const handleFilterSelection = (
+    option: SelectOption,
+    event?: React.SyntheticEvent<HTMLElement, Event> | undefined
+  ) => {
+    event?.stopPropagation()
+    event?.preventDefault()
+
+    if (option.value) {
+      updateQueryParams({
+        filterIdentifier: option.value.toString(),
+        filters: [] as any /* this will remove the param */
+      })
+    } else {
+      updateQueryParams({
+        filterIdentifier: [] as any /* this will remove the param */,
+        filters: [] as any /* this will remove the param */
+      })
+    }
+  }
+
+  const onApply = (inputFormData: FormikProps<EnvironmentGroupFilterFormType>['values']) => {
+    if (!isObjectEmpty(inputFormData)) {
+      const filterFromFormData = {
+        ...(inputFormData.envGroupName && {
+          envGroupName: inputFormData.envGroupName
+        }),
+        ...(inputFormData.description && {
+          description: inputFormData.description
+        }),
+        ...(inputFormData.envGroupTags && {
+          envGroupTags: Object.keys(inputFormData.envGroupTags || {})?.map((key: string) => {
+            return { key, value: inputFormData?.envGroupTags?.[key] } as NGTag
+          })
+        }),
+        ...(inputFormData.environments?.length && {
+          envIdentifiers: inputFormData.environments?.map((env: MultiSelectOption) => env?.value)
+        })
+      }
+      updateQueryParams({ page: [] as any, filters: JSON.stringify({ ...(filterFromFormData || {}) }) })
```

Comment:
Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9264786)
```suggestion
      updateQueryParams({ page: [] as unknown, filters: JSON.stringify({ ...(filterFromFormData || {}) }) })
```